spec/services/commitments/calculate_amount_service_spec.rb

```diff
@@ -186,12 +208,23 @@
         context 'when a commitment exists for a plan' do
           let(:commitment) { create(:commitment, plan:, amount_cents:) }
 
-          it { is_expected.to delegate_method(:subscription).to(:invoice_subscription) }
+          context 'when it is full period' do
+            it 'returns result' do
+              result = apply_service.call
 
-          it 'returns result' do
-            result = apply_service.call
+              expect(result.commitment_amount_cents).to eq(commitment.amount_cents)
+            end
+          end
 
-            expect(result.commitment_amount_cents).to eq(commitment.amount_cents)
+          context 'when it is not full period' do
+            let(:to_datetime) { DateTime.parse('2024-03-30T23:59:59') }
+            let(:timestamp) { DateTime.parse('2024-04-04T10:00:00') }
+
+            it 'returns prorated result' do
+              result = apply_service.call
+
+              expect(result.commitment_amount_cents).to eq(39560)
```

Comment:
Style/NumericLiterals: Use underscores(_) as thousands separator and separate every 3 digits with them. (https://rubystyle.guide#underscores-in-numerics)